### PR TITLE
[VideoGallery] fix logic for state strings

### PR DESCRIPTION
**Pull request by dmceachernmsft**
# What

Updates state logic for 'Connecting' and 'Ringing' participant state strings
# Why

Statement would only eval to 'Connecting' state

https://skype.visualstudio.com/SPOOL/_workitems/edit/3914972
# How Tested

Snapshots to show diff from old logic

### CODE REVIEW

**Review by mgamis-msft — APPROVED** [review]

File: packages/react-components/src/components/VideoGallery/utils/videoGalleryLayoutUtils.ts [hunk 1]
```diff
@@ -89,8 +89,12 @@ const getOrganizedParticipants = (props: OrganizedParticipantsArgs): OrganizedPa
 
   let gridParticipants = newGridParticipants;
   let overflowGalleryParticipants = newOverflowGalleryParticipants;
-  if (gridParticipants.length + callingParticipants.length <= maxGridParticipants) {
-    gridParticipants = gridParticipants.concat(callingParticipants);
+  if (overflowGalleryParticipants.length === 0) {
+    const numberOfCallingParticipantsInGrid = maxGridParticipants - gridParticipants.length;
+    const gridCallingParticipants = callingParticipants.slice(0, numberOfCallingParticipantsInGrid);
+    const overflowGalleryCallingParticipants = callingParticipants.slice(numberOfCallingParticipantsInGrid);
+    gridParticipants = gridParticipants.concat(gridCallingParticipants);
```
Comment (JamesBurnside) [inline packages/react-components/src/components/VideoGallery/utils/videoGalleryLayoutUtils.ts:96]: It's really hard to parse when reading the difference between `participantsForGrid`, `newGridParticipants`, `gridParticipants` and `gridCallingParticipants`, is there anything we can do to improve this chunk?

File: packages/react-components/src/components/VideoGallery/utils/videoGalleryLayoutUtils.ts [hunk 2]
```diff
@@ -89,8 +89,12 @@ const getOrganizedParticipants = (props: OrganizedParticipantsArgs): OrganizedPa
 
   let gridParticipants = newGridParticipants;
   let overflowGalleryParticipants = newOverflowGalleryParticipants;
```
Comment (JamesBurnside) [inline packages/react-components/src/components/VideoGallery/utils/videoGalleryLayoutUtils.ts:91]: We immediately overwrite `overflowGalleryParticipants` in the if-else. It reassigns to either `overflowGalleryCallingParticipants` or `overflowGalleryParticipants.concat(callingParticipants);`

so it looks like the value of newOverflowGalleryParticipants is being thrown out unused?

Comment (JamesBurnside) [reply packages/react-components/src/components/VideoGallery/utils/videoGalleryLayoutUtils.ts:91]: Oh wait in the else it concats with gridCallingParticipants

File: packages/react-components/src/components/VideoGallery/utils/videoGalleryLayoutUtils.ts (hunk 1 above)
Comment (JamesBurnside) [reply packages/react-components/src/components/VideoGallery/utils/videoGalleryLayoutUtils.ts:96]: Perhaps even just a comment like:
```
// Add the participants being called into the call.
// If there are already overflow participants, add these to the array of overflow participants
// Otherwise, add the maximum number to the main grid, then put the rest in the overflow
```

**Review by JamesBurnside — APPROVED** [review]
